lms/views/api/vitalsource.py

```diff
@@ -11,33 +11,23 @@
 class VitalSourceAPIViews:
     def __init__(self, request):
         self.request = request
-        self.vitalsource_service = request.find_service(VitalSourceService)
+        self.svc = request.find_service(VitalSourceService)
 
     @view_config(route_name="vitalsource_api.books.info")
     def book_info(self):
-        book_id = self._get_book_id()
-
-        book_info = self.vitalsource_service.get_book_info(book_id)
-        return {
-            "id": book_info["vbid"],
-            "title": book_info["title"],
-            "cover_image": book_info["resource_links"]["cover_image"],
-        }
+        return self.svc.get_book_info(self._book_id)
 
     @view_config(route_name="vitalsource_api.books.toc")
     def table_of_contents(self):
-        book_id = self._get_book_id()
+        return self.svc.get_table_of_contents(self._book_id)
 
-        book_toc = self.vitalsource_service.get_book_toc(book_id)
-        return book_toc["table_of_contents"]
+    _VALID_BOOK_ID = re.compile(r"^[\dA-Z-]+$")
 
-    def _get_book_id(self):
+    @property
+    def _book_id(self):
         book_id = self.request.matchdict["book_id"]
-        if not self._is_valid_book_id(book_id):
+
+        if not self._VALID_BOOK_ID.match(book_id):
             raise HTTPBadRequest("Invalid `book_id`. It must only contain [0-9A-Z-].")
```

Comment:
I'd like to do this in the VS client, but then I'm not sure what error we'd raise. If we had to catch it here again, it's probably a waste of time.

Reply:
Turns out the answer was to apply a schema here, then once the perimeter is controlled, everything else can assume this is valid.

Not the only way to go about it, but it matches what we do elsewhere and is simple to implement.

Unfortunately this means we can't unit test this any more. I tried adding functional tests, but there's too much missing machinery on that side to do this easily. I got into messing about trying to generate valid auth tokens and eventually gave up.

It might be good to one day crack that nut so we can test these end-points without huge amounts of effort, but the first will be hard.